clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/recordreader/avro/AbstractAvroRecordReader.java

```diff
@@ -60,7 +61,21 @@ public AbstractAvroRecordReader(
         if (fileField.name().equals(valueFieldStr)) {
           fieldSchema = ETLUtils.getValueSchemaFromETLValueSchema(fieldSchema, etlValueSchemaTransformation);
         }
-        outputSchemaFields.add(AvroCompatibilityHelper.newField(fileField).setSchema(fieldSchema).build());
+        // Since we are stripping the schema of the value field of the "null" type, we need to handle the case where the
+        // default value of the schema is null and the schema is not nullable. In this case, we create a schema without
```

Comment:
I'm not sure I understand this one.  So we stripped out the null type, but this right now chokes if the default value is null?  Is that right?

Reply:
Yes, exactly. The problem arises because when we do `AvroCompatibilityHelper.newField(fileField)`, then it copies all properties of `fileField`